File: Perfect crossover/crossover.cpp
//#include <bits/stdc++.h>
#include <iostream>
#include <vector>
#include <string>

using namespace std;

string crossover(string s, string t){
	
	if (s == "")
		return t;
	if (t == "")
		return s;
	
	int lens = s.size();
	int lent = t.size();
	
	string firsts = s.substr(0, 1);
	string rests = s.substr(1, lens);
	string lastt = t.substr(lent - 1, 1);
	string restt = t.substr(0, lent - 1);
	
	if (firsts == lastt){
		return crossover(restt, rests);
	} else {
		return lastt + crossover(rests, restt) + firsts;
	}
}

int main(){
	
//	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
	
	string s, t;
	
	cin >> s >> t;
	
//	cout << crossover(s, t) << "\n";
	
	
	string prima = "";
	string dopo = "";
	
	while (s != "" && t != ""){
		int lens = s.size();
		int lent = t.size();
		
		string firsts = s.substr(0, 1);
		string rests = s.substr(1, lens);
		string lastt = t.substr(lent - 1, 1);
		string restt = t.substr(0, lent - 1);
		
		if (firsts == lastt){
			s = restt;
			t = rests;
		} else {
			prima = prima + lastt;
			dopo = firsts + dopo;
			s = rests;
			t = restt;
		}
		
	}
	
	cout << prima + s + t + dopo << "\n";
	
	return 0;
}

File: Perfect crossover/crossover.py
#Scrivi qui il tuo codice


def crossover(s, t):
    
    if s == "":
        return t
    elif t == "":
        return s
    
    lens = len(s)
    lent = len(t)
    
    firsts = s[:1]
    rests = s[1:]
    lastt = t[lent - 1:]
    restt = t[:lent - 1]
    
    if firsts == lastt:
        
        return crossover(restt, rests)
    
    else:
        
        return lastt + crossover(rests, restt) + firsts
    
    
    
    
    
    
    
    


s = str(input().strip())
t = str(input().strip())



#print("ricorsiva: ", crossover(s, t))



prima = ""
dopo = ""
#print(s, t)
while s != "" and t != "":
    lens = len(s)
    lent = len(t)
    
    firsts = s[:1]
    rests = s[1:]
    lastt = t[lent - 1:]
    restt = t[:lent - 1]
    
    #print("ciao")
    
    if firsts == lastt:
        t = rests
        s = restt
    else:
        prima = prima + lastt
        dopo = firsts + dopo
        s = rests
        t = restt
    
    #print(prima, s, t, dopo)


print(prima + s + t + dopo)



a = "ciao"
b = "a"


#print (a[2:])
#print(b[0:])
#print(b[1:] == "")
